plugin/src/main/java/io/jenkins/plugins/coverage/model/FileCoverageNode.java

```diff
@@ -16,6 +21,11 @@
 
     private final String sourcePath;
 
+    // new since 3.0.0
+    private SortedMap<Integer, Coverage> coveragePerLine = new TreeMap<>();
```

Comment:
Shouldn't we store the type as well? I.e. for line 4 we have in the JaCoCo report: 
```
<line nr="4" mi="2" ci="14" mb="1" cb="5"/>
```

This should produce a 
- Line Coverage of 1 / 1
- Instruction Coverage of 14 / 16
- Branch Coverage of 5 / 6

Reply:
At the moment, the stored `Coverage` provides information about line- and branch-coverage. In my opinion, instruction-coverage is not required here, since I guess in the most cases, it is not used.
Anyways, I would like to refactor this at all and store the `coverageLineMapping` within `MethodCoverageNode` since at the moment, the coverage mapping is not applyable on methods. The required storage would be the same and at the same time, we can use the lines numbers per method within the mapping for filtering source code for specific methods in further releases.
If instruction-coverage is required and worth the additional storage, I can wrap the `Coverage` together with its metric when I apply the described refactoring. 
What is your opinion on this?

Reply:
No this is not required. I don't use instruction coverage on my own. So I am fine with just recording line and branch coverage. But even if we do not use instruction coverage there should be values for line *and* branch coverage. Or are you storing one tree with line coverage and one for branch coverage?

Reply:
No, there is just one `Coverage` for each line. The information "line or branch" is given automatically by `getTotal()`, since line coverage has the value `1` and branch coverage has `>1`.
The logic I wrote for calculating coverage deltas distinguishes this already.
The reason I decided this is that having two `Coverage` per line would double the required space, which means that for a project with 100.000 LOC the XML would be nearly 100.000 Coverage-Tags larger.

Reply:
I see! Then please add a small JavaDoc comment for that field that describes that modeling.